### DISCUSSION TITLE
Controlling string parsing

### DISCUSSION THREAD

**Discussion by gsakkis** [hgrecco/pint#1885] [Q&A]
Pint comes with a powerful string parsing that can parse almost arbitrarily complex expressions. In my use case I'd like to limit parsing to a simple "number [unit]" pattern. For example 
- "3-8" represents a range instead of a quantity and should raise an exception instead of evaluating to -5.
- "6 +- 1" represents an uncertainty instead of a quantity and should raise an exception instead of evaluating to 5.
- "3/4" represents a fraction and should still be allowed to be parsed as quantity. 

Is this possible somehow?

**Comment by hgrecco** [comment]
For simple things, you can use `parse_pattern` (See [here](https://pint.readthedocs.io/en/stable/getting/tutorial.html#string-parsing) ). For more complex things, maybe my [stringparser](https://github.com/hgrecco/stringparser/) package might be useful for you.

**Comment by gsakkis** [comment]
Thanks, I looked at `parse_pattern` but it seems it expects a given unit? Or is there a way to parse arbitrary units?

For now I resorted to checking if a regular expression matches before trying to parse it as a quantity.

**Reply by hgrecco** [reply]
`parse_pattern` is used to capture known parts of a string, which you can then use to build your own Quantity.
